File: LeetCode/0905. Sort Array By Parity/solution.cpp
/**
 * 285 / 285 test cases passed.
 * Runtime: 12 ms
 * Memory Usage: 15.7 MB 
 */
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& nums) {
        int l = 0;       
        for (int r = 0; r < nums.size(); ++ r) {
            if (nums[r] % 2 == 0) {
                swap(nums[l], nums[r]);
                ++ l;
            }
        }
        return nums;
    }
};

File: LeetCode/0905. Sort Array By Parity/solution.py
"""
285 / 285 test cases passed.
Runtime: 32 ms
Memory Usage: 15.6 MB
"""
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        l = 0
        for r in range(len(nums)):
            if nums[r] % 2 == 0:
                nums[l], nums[r] = nums[r], nums[l]
                l += 1
        return nums
